solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -763,8 +764,53 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
         validAddress(newProvider)
         returns (uint256)
     {
+        return transferPosition(msg.sender, id, newProvider);
+    }
+
+    /**
+     * @dev transfers a position to a new provider and optionally notifies another contract
+     *
+     * @param id position id
+     * @param newProvider the new provider
+     * @param target the contract to notify
+     * @param data the data to call the contract with
+     *
+     * @return new position id
+     */
+    function transferPositionAndCall(
+        uint256 id,
+        address newProvider,
+        address target,
+        bytes memory data
+    ) external protected validAddress(newProvider) returns (uint256) {
+        uint256 newId = transferPosition(msg.sender, id, newProvider);
+
+        // make sure that we're not trying to call into the zero address or a fallback function
+        require(target != address(0) && data.length > 0, "ERR_INVALID_CALL_DATA");
```

Comment:
Alternative 1:
```
        validAddress(target);
        ...
        require(data.length > 0, "ERR_INVALID_CALL_DATA");
```

Alternative 2:
```
        require(target != address(0), "ERR_INVALID_CALL_TARGET");
        require(data.length > 0, "ERR_INVALID_CALL_DATA");
```

Reply:
Also, it's possibly better to check `data.length >= 4` (length of function selector), but I'm not sure about this.

Reply:
Regarding the alternatives, I'm not sure there's any benefit in separating this test into two tests. I will extend the `data.length`, but keep in mind that we might need to remove it entirely if we'll decide to support fallback functions.